test/lib/course_planner/offered_courses_test.exs

```diff
@@ -73,4 +73,135 @@ defmodule CoursePlanner.OfferedCoursesTest do
+
+      assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances()
+    end
+
+    test "when all attendances of the class are filled for an offered_course" do
+     students = insert_list(3, :student)
+     teacher = insert(:teacher)
+     class = insert(:class, date: Timex.now())
+     Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class, attendance_type: "Present")
+     end)
+     insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+     requested_current_date =  Timex.shift(Timex.now(), days: 2)
+     assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances(requested_current_date)
+    end
+
+    test "when there is missing attendances for one class of an offered_course" do
+     students = insert_list(3, :student)
+     teacher = insert(:teacher)
+     class = insert(:class, date: Timex.now())
+     Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class, attendance_type: "Not filled")
+     end)
+     offered_course =
+       insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+     requested_current_date =  Timex.shift(Timex.now(), days: 2)
+     [not_filled_offered_course] = OfferedCourses.get_offered_courses_with_not_filled_class_attendances(requested_current_date)
+     assert offered_course.id == not_filled_offered_course.id
+    end
+
+    test "when there is missing attendances for some classes of an offered_course" do
+      students = insert_list(3, :student)
+      teacher = insert(:teacher)
+      [class1, class2, class3] = insert_list(3, :class, date: Timex.now())
+
+      Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class1, attendance_type: "Not filled")
+      end)
+      Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class2, attendance_type: "Present")
+      end)
+
+      [student1, student2, student3] = students
+      insert(:attendance, student: student1, class: class3, attendance_type: "Present")
+      insert(:attendance, student: student2, class: class3, attendance_type: "Not filled")
+      insert(:attendance, student: student3, class: class3, attendance_type: "Present")
+
+      offered_course =
+       insert(:offered_course, classes: [class1, class2, class3], students: students, teachers: [teacher])
+
+      requested_current_date =  Timex.shift(Timex.now(), days: 2)
+      [not_filled_offered_course] = OfferedCourses.get_offered_courses_with_not_filled_class_attendances(requested_current_date)
+
+      not_filled_classes =
+        not_filled_offered_course.classes
+        |> Enum.map(&(&1.id))
+        |> Enum.sort(&(&1 < &2))
+      expected_classes =
+        [class1.id, class3.id]
+        |> Enum.sort(&(&1 < &2))
+
+      assert offered_course.id == not_filled_offered_course.id
+      assert expected_classes == not_filled_classes
```

Comment:
what do you think of defining `not_filled_classes` as
```elixir
not_filled_classes = Enum.map(not_filled_offered_course.classes, &(&1.id))
```
then assert
```elixir
assert class1.id in not_filled_classes
assert class3.id in not_filled_classes
```
that's easier to understand in my opinion